cloud/auth/auth.go

```diff
@@ -301,25 +302,42 @@ func switchToLastUsedWorkspace(c *config.Context, workspaces []astro.Workspace)
 	return astro.Workspace{}, false, nil
 }
 
-// checkToken requests a users rolebindings and sets the workspace to make sure that token works
-// TODO check orgID is in the token
-func checkToken(c *config.Context, client astro.Client, out io.Writer) error {
-	self, err := client.GetUserInfo()
+// check client status after a successfully login
+func checkUserSession(c *config.Context, client astro.Client, coreClient astrocore.CoreClient, out io.Writer) error {
+	// fetch self user base on token
+	// we set CreateIfNotExist to true so we always create astro user when a successfully login
+	createIfNotExist := true
+	selfResp, err := coreClient.GetSelfUserWithResponse(http_context.Background(), &astrocore.GetSelfUserParams{
+		CreateIfNotExist: &createIfNotExist,
+	})
+	err = astrocore.NormalizeAPIError(selfResp.HTTPResponse, selfResp.Body, err)
 	if err != nil {
 		return err
 	}
-	organizationID := self.AuthenticatedOrganizationID
-
-	if organizationID != "" {
-		err = c.SetContextKey("organization", organizationID)
-		if err != nil {
-			return err
+	authOrgID := *selfResp.JSON200.OrganizationId
+	// fetch all orgs that the user can access
+	orgsResp, err := coreClient.ListOrganizationsWithResponse(http_context.Background())
+	err = astrocore.NormalizeAPIError(orgsResp.HTTPResponse, orgsResp.Body, err)
+	if err != nil {
+		return err
+	}
+	orgs := *orgsResp.JSON200
+	// default to first one in case something crazy happen lol
+	activeOrg := orgs[0]
+	for i := range orgs {
+		if orgs[i].Id == authOrgID {
+			activeOrg = orgs[i]
+			break
 		}
 	}
+	err = c.SetOrganizationContext(activeOrg.Id, activeOrg.ShortName)
```

Comment:
This is guard against a rare use case where user's login org is different than the org he can access. We always treat AuthZ is the source of truth of user org relationship